homeassistant/components/esphome/sensor.py

```diff
@@ -52,9 +58,22 @@ def _state(self) -> Optional[SensorState]:
     @property
     def icon(self) -> str:
         """Return the icon."""
-        if not self._static_info.icon or self._static_info.device_class:
+        if self._static_info.device_class:
             return None
-        return self._static_info.icon
+        icon_validator = vol.Or(cv.whitespace, cv.icon)
```

Comment:
Why do we use the whitespace validator?

Reply:
I took that code from the validation of the icon attribute of the template sensor: https://github.com/home-assistant/core/blob/b2df9aaaf1060916e8484260d0914ec3872581e3/homeassistant/components/template/template_entity.py#L278

However, I removed the icon string validation as suggested in your second comment.